keyberon/src/layout.rs

```diff
@@ -71,6 +73,8 @@ pub struct Layout<'a, const C: usize, const R: usize, const L: usize, T = core::
 where
     T: 'a + std::fmt::Debug,
 {
+    /// Fallback for transparent keys inside actions that are on `default_layer`.
+    pub src_keys: &'a [Action<'a, T>; C],
```

Comment:
What necessitated this change? I thought that the parser layer handled this already, but maybe there's a deficiency there that should be fixed instead?

If it can be removed, I would prefer to not include this change since it has memory+code overhead that seem like it could be avoided.

Reply:
Yes in parser it's implemented that `_` on base layer are replaced to key from defsrc. But it doesn't apply to `_` on base layer that are inside actions.

Speaking about cpu overhead, is this change really adding that much overhead? All overhead that's added here is literally just one function call and one conditional statement. I haven't benchmarked this, but my guess is that it will add <0.01ms latency.

Considering this doesn't add much cpu overhead, why not optimize for memory, by removing a second copy of each layer and using the same approach I've proposed here for all Trans actions? This would remove a lot of code complexity and a lot of memory usage (my guess is 40% decrease)

Reply:
Makes sense!

Reply:
Okay! I've created #891 to track this

Reply:
For now can we go for current (dual) approach to resolvin Trans keys on base layer? The proposed change to refactor to use only dynamic resolution approach would go in other PR.

Reply:
Yep I'm fine leaving it as-is for now, since I understand the context.